graphene/src/document.rs

```diff
@@ -485,6 +485,17 @@ impl Document {
 				self.set_layer(path, Layer::new(LayerDataType::Shape(Shape::poly_line(points, *style)), *transform), *insert_index)?;
 				Some([vec![DocumentChanged, CreatedLayer { path: path.clone() }], update_thumbnails_upstream(path)].concat())
 			}
+			Operation::AddFreehand {
+				path,
+				insert_index,
+				points,
+				transform,
+				style,
+			} => {
+				let points: Vec<glam::DVec2> = points.iter().map(|&it| it.into()).collect();
+				self.set_layer(path, Layer::new(LayerDataType::Shape(Shape::poly_line(points, *style)), *transform), *insert_index)?;
+				Some([vec![DocumentChanged, CreatedLayer { path: path.clone() }], update_thumbnails_upstream(path)].concat())
+			}
```

Comment:
This is duplicated behaviour. Consider (for now) merging this with the `Operation::AddPen` and calling it `Operation::AddPolyline` or similar to reflect the actual shape type that is created.

Reply:
Consolidated this to Operation::AddPolyline as suggested.